library/core/class.format.php

```diff
@@ -1713,6 +1714,49 @@ public static function mentions($Mixed) {
         }
     }
 
+    /**
+     * Optionally replace any embedded images from untrusted sites with links to the image.
+     *
+     * @param string $html String from which you might want to strip out untrusted images.
+     * @return string The string with images replaced by links.
+     */
+    public static function sanitizeImages($html) {
+        // If the config does not have a SanitizeImages entry or it is set to false, pass the HTML to the renderer as is.
+        if (!c('Garden.Format.SanitizeImages', false)) {
+            return $html;
+        }
+
+        preg_match_all('/\<img\s+src\s*=\s*[\"\'](.*?)[\"\'].*\>/', $html, $matches, PREG_SET_ORDER);
+        // If no images are found in the HTML, pass the HTML to the renderer as is.
+        if (!$matches) {
+            return $html;
+        }
+
+        foreach ($matches as $image)  {
+            $imageUrl = parse_url($image[1]);
+            $stripImage = false;
+
+            // If the image is not coming from this forum's CDN, strip this image.
+            if (c('S3.Prefix') && c('S3.Zone')) {
+                if ($imageUrl['host'] !== c('S3.Zone').".v-cdn.net" && !stringBeginsWith($imageUrl['path'], '/'.c('S3.Prefix'))) {
+                    $stripImage = true;
+                }
+            }
+
+            // If the host of the image source is a trusted domain, strip this image
+            if (!in_array(val('host', $imageUrl), trustedDomains())) {
+                $stripImage = true;
+            }
+
+            // Replace the image embed tag with a link to the image.
+            if ($stripImage) {
+                $imageName = trim(substr(val('path', $imageUrl), strrpos(val('path', $imageUrl), '\/')), '/');
+                $html = preg_replace('/\<img\s+src\s*=\s*[\"\']('.preg_quote($image[1], '/').')[\"\'].*\>/', '<a href="$1" target="_blank" title="'.t('Image from unconfirmed host').'">'.$imageName.'</a>', $html);
```

Comment:
I do not think that title helps, and in any case it would need to be added to locales if it was.

I don't think we want to hard-code `target="_blank"` either.